cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -309,6 +309,15 @@ if ("${MAKE}" STREQUAL "")
     endif()
 endif()
 
+# Using make -j in sub-make is fragile
+# see discussion https://github.com/apache/arrow/pull/2779
+if (${CMAKE_GENERATOR} MATCHES "Makefiles")
+    set(MAKE_BUILD_ARGS "")
+else()
+    # limit the maximum number of jobs for ninja
```

Comment:
You may misunderstand. This argument is used by `make` not `ninja`.

Reply:
@kou No, I don't fully understand :)
My conclusion from your reasoning and patch was to not parallelize `make` at all, and parallelize but limit the number of jobs for `ninja -j4`. 

BTW The build is passing with the current changeset.

Reply:
Right.